public/JobHandler/index.ts

```diff
@@ -0,0 +1,92 @@
+import { CoreSetup } from "../../../../src/core/public";
+import { jobSchedulerInstance } from "../context/JobSchedulerContext";
+import { IndexService } from "../services";
+import { ReindexJobMetaData, RecoveryJobMetaData } from "../models/interfaces";
+
+export function JobHandlerRegister(core: CoreSetup) {
+  const indexService = new IndexService(core.http);
+  jobSchedulerInstance.addCallback({
+    callbackName: "callbackForReindex",
+    callback: async (job: ReindexJobMetaData) => {
+      const extras = job.extras;
+      const indexResult = await indexService.getIndices({
+        from: 0,
+        size: 10,
+        search: extras.destIndex,
+        terms: extras.destIndex,
+        sortField: "index",
+        sortDirection: "desc",
+        showDataStreams: extras.isDataStream || false,
+      });
+      if (indexResult.ok) {
+        const [firstItem] = indexResult.response.indices || [];
+        if (firstItem && firstItem.status !== "reindex") {
+          core.notifications.toasts.addSuccess(
+            `Reindex from [${extras.sourceIndex}] to [${extras.destIndex}] has been finished successfully.`,
+            {
+              toastLifeTimeMs: 1000 * 60 * 60 * 24 * 5,
+            }
+          );
+          return true;
+        }
+      }
```

Comment:
this will not tell whether reindex is success or failed, the difference for success and fail is the result of task result, if we have error in task result, it means there has some errors. for example 

```
 "error" : {
      "type" : "action_request_validation_exception",
      "reason" : "Validation Failed: 1: reindex cannot write into an index its reading from [.ds-redis-logs-monthly-000001];"
    }
```

Reply:
Yes, you are right and there is issue for recovery check. Let me check with both of you to adjust my logic.